package com.gogidix.rapidassist.metrics.telemetry.service.infrastructure.web;

import jakarta.validation.constraints.NotBlank;

import java.util.Map;

public class IngestTelemetryEventRequest {

    @NotBlank
    private String type;

    @NotBlank
    private String name;

    private Double value;

    private String correlationId;

    private Map<String, String> attributes;

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Double getValue() {
        return value;
    }

    public void setValue(Double value) {
        this.value = value;
    }

    public String getCorrelationId() {
        return correlationId;
    }

    public void setCorrelationId(String correlationId) {
        this.correlationId = correlationId;
    }

    public Map<String, String> getAttributes() {
        return attributes;
    }

    public void setAttributes(Map<String, String> attributes) {
        this.attributes = attributes;
    }
}
